### PR TITLE
Add compliant and noncompliant examples of java/cookie-httponly@v1.0

### PR DESCRIPTION
**Pull request by prajaki**
*Description of changes:*
* Rules Added: java/cookie-httponly@v1.0
* Added new methods for compliant and non compliant examples
* Added new class
* Added dependencies on Spring Framework

### CODE REVIEW

File: src/java/detectors/cookie_httponly/CookieHttpOnly.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+package detectors.cookie_httponly_rule;
+
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.Cookie;
+
+public class CookieHttpOnly{
+    // {fact rule=cookie-httponly@v1.0 defects=1}
+    public void CookieHttpOnlyNoncompliant(HttpServletResponse res) {
+        Cookie c = new Cookie("name", "not-null");
```
Comment (zjiamzn) [inline src/java/detectors/cookie_httponly/CookieHttpOnly.java:14]: just call it `cookie`

File: src/java/detectors/cookie_httponly/CookieHttpOnly.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        c.setSecure(true);
+        // Noncompliant: HttpOnly is not set before adding the cookie.
+        res.addCookie(c);
+    }
+    // {/fact}
+
+    // {fact rule=cookie-httponly@v1.0 defects=0}
+    public void CookieHttpOnlyCompliant(HttpServletResponse res) {
+        Cookie c = new Cookie("name", "not-null");
```
Comment (zjiamzn) [inline src/java/detectors/cookie_httponly/CookieHttpOnly.java:23]: `cookie` is more expressive

File: src/java/detectors/cookie_httponly/CookieHttpOnly.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ */
+
+package detectors.cookie_httponly_rule;
+
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.Cookie;
+
+public class CookieHttpOnly{
+    // {fact rule=cookie-httponly@v1.0 defects=1}
+    public void CookieHttpOnlyNoncompliant(HttpServletResponse res) {
```
Comment (zjiamzn) [inline src/java/detectors/cookie_httponly/CookieHttpOnly.java:13]: method name should not start with a capital letter: `cookieHttpOnlyNoncompliant`

File: src/java/detectors/cookie_httponly/CookieHttpOnly.java (hunk 2 above)
Comment (prajaki) [reply src/java/detectors/cookie_httponly/CookieHttpOnly.java]: Renamed the variable.

File: src/java/detectors/cookie_httponly/CookieHttpOnly.java (hunk 3 above)
Comment (prajaki) [reply src/java/detectors/cookie_httponly/CookieHttpOnly.java]: Renamed the method name

**Comment by prajaki** [comment]
Here already existing manifest - **_java/insecure-cookie@v1.0_** is  more correct . Thus new manifest is not needed, hence closing this PR.

**Pull request closed without merging**